# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStore.h
# Commit: 0868fb4018a7
# Full Hash: 0868fb4018a7ae2002b8f2fcfd88643c2e060f07
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 9c8b6e35bf3f -r 0868fb4018a7 dom/cookiestore/CookieStore.h
--- a/dom/cookiestore/CookieStore.h	Tue Sep 10 16:31:38 2024 +0000
+++ b/dom/cookiestore/CookieStore.h	Tue Sep 10 16:31:38 2024 +0000
@@ -15,6 +15,9 @@
 
 namespace mozilla::dom {
 
+class CookieData;
+class CookieStoreChild;
+class CookieStoreNotifier;
 class Promise;
 
 class CookieStore final : public DOMEventTargetHelper {
@@ -29,6 +32,8 @@
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
+  void FireDelayedDOMEvents();
+
   already_AddRefed<Promise> Get(const nsAString& aName, ErrorResult& aRv);
 
   already_AddRefed<Promise> Get(const CookieStoreGetOptions& aOptions,
@@ -54,6 +59,19 @@
  private:
   explicit CookieStore(nsIGlobalObject* aGlobal);
   ~CookieStore();
+
+  void Shutdown();
+
+  Document* MaybeGetDocument() const;
+
+  already_AddRefed<Promise> GetInternal(const CookieStoreGetOptions& aOptions,
+                                        bool aOnlyTheFirstMatch,
+                                        ErrorResult& aRv);
+
+  bool MaybeCreateActor();
+
+  RefPtr<CookieStoreChild> mActor;
+  RefPtr<CookieStoreNotifier> mNotifier;
 };
 
 }  // namespace mozilla::dom